pkg/k8s/k8s.go

```diff
@@ -81,7 +75,21 @@ func New(ctx context.Context, namespace string) (*Client, error) {
 	return kc, nil
 }
 
-func (c *Client) Clientset() *kubernetes.Clientset {
+func NewCustom(
```

Comment:
Few comments on naming conventions. 

`NewCustom` implies there is a `New` function that could be called that returns some default. 
Since there aren't defaults, and what is currently being supplied are the minimum requirements, I would call this `NewClient`.  

We should also always include the name of what is being returned to be explicit. Especially since our `k8` package has a bunch of structs defined that could have `New` methods.

Reply:
good point.

Reply:
well I think the return type is disruptive enough:

```go
func New(ctx context.Context, namespace string) (*Client, error)
```
there is only one struct called `Client` .
would you mind to elaborate more on this point?

Reply:
and that `New` fucntion initilizes the required values by itself. and that `NewCustom` does not. which is used mostly for tests:
```go
func NewCustom(
	cs kubernetes.Interface,
	dc discovery.DiscoveryInterface,
	dC dynamic.Interface,
	namespace string,
) *Client 
```

so how do you think we can make this more clear?

Reply:
We shouldn't rely on return types as the descriptor because that isn't what people are reading first.  Developers are going to be typing out `k8.New...`.  When naming we should think about the developer using the method, not us, the developers building it.  1k8.New` and `k8.NewCustom` aren't clear enough imo. 

The k8 package has lots of different types which is why we need the descriptor in the name, otherwise no other type could have a `New` method. 

This is how `New` and `NewCustom` methods should interact
```go 
func NewClient(ctx context.Context, namespace string) (*Client, error) {
	config, err := getClusterConfig()
	if err != nil {
		return nil, ErrRetrievingKubernetesConfig.Wrap(err)
	}

	cs, err := kubernetes.NewForConfig(config)
	if err != nil {
		return nil, ErrCreatingClientset.Wrap(err)
	}

	// create discovery client
	dc, err := discovery.NewDiscoveryClientForConfig(config)
	if err != nil {
		return nil, ErrCreatingDiscoveryClient.Wrap(err)
	}

	// Create the dynamic client
	dC, err := dynamic.NewForConfig(config)
	if err != nil {
		return nil, ErrCreatingDynamicClient.Wrap(err)
	}
	return NewClientCustom(cs, dc, dC, namespace)
}

func NewClientCustom(
	cs kubernetes.Interface,
	dc discovery.DiscoveryInterface,
	dC dynamic.Interface,
	namespace string,
) (*Client, error) {
	kc := &Client{clientset: cs, discoveryClient: dc, dynamicClient: dC}

	namespace = SanitizeName(namespace)
	kc.namespace = namespace
	if kc.NamespaceExists(ctx, namespace) {
		logrus.Debugf("Namespace %s already exists, continuing.\n", namespace)
		return kc, nil
	}

	if err := kc.CreateNamespace(ctx, namespace); err != nil {
		return nil, ErrCreatingNamespace.WithParams(namespace).Wrap(err)
	}

	return kc, nil
}
```